### PR TITLE
chore: use poetry

### PR DESCRIPTION
**Pull request by thalesmg**
Recently we had an [incident](https://github.com/emqx/cdk-emqx-cluster/pull/94) with unpinned package versions.

To void that, we may start using [Poetry](https://python-poetry.org/docs/basic-usage/) for managing the dependencies.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
+RUN curl -sSL https://install.python-poetry.org | POETRY_VERSION=1.1.14 python3 -
+ENV PATH=/root/.local/bin:$PATH
+WORKDIR /setup
+COPY pyproject.toml /setup/pyproject.toml
+COPY poetry.lock /setup/poetry.lock
+# install only cdk-emqx-cluster's deps for better docker caching
+RUN poetry install --no-root
 COPY README.md /setup/
 RUN mkdir -p /setup/cdk_emqx_cluster
-RUN cd /setup/ \
```
Comment (savonarola) [inline Dockerfile:9]: I am not a Python pro but IMO with poetry the flow is quite the same as with `requirements.txt`
We
* Make venv
* Activate venv
* Install poetry inside venv with pip (`pip install poetry`)
* Having poetry and poetry.lock, now (inside venv) we run `poetry install` to have locked packages installed.

Comment (thalesmg) [reply Dockerfile:9]: I'll try that alternative, but it seemed to me during the tests that it installed things only in its own virtual env, not the current one or globally.  And that needs to be activated with `poetry shell`.

But I'll try that as well to confirm.

Comment (savonarola) [reply Dockerfile:9]: Some pythonistas use poetrty to manage venvs, but IMO this brings unnecessary complexity 😅

Comment (thalesmg) [reply Dockerfile:9]: Indeed, that worked.  I must've made some mistake while testing :see_no_evil: 

https://github.com/emqx/cdk-emqx-cluster/pull/96/commits/27b73f5d9b19c5a542beccca41eb171b7acf0911

**Comment by qzhuyan** [comment]
I think we should ask cdk project if this is the right approach , the old method is generated from cdk template which should be stable.

**Comment by thalesmg** [comment]
> I think we should ask cdk project if this is the right approach , the old method is generated from cdk template which should be stable.

It does suffer from the reproducibility issue that @savonarola has hit yesterday: when he built a new docker image with the partially pinned dependencies, it wouldn't run.  So I guess that we either would need to i) keep updating the dependencies, or ii) properly pin them all of them, which would require something like `pip freeze > requirements.txt`.  :thinking: 

I think that (i) is more stable, since (ii) already involves changing the default `requirements.txt` (which contains only `-e .`)

**Comment by thalesmg** [comment]
superseded by #99

**Pull request closed without merging**